### PR TITLE
Fallback matmul codegen to TileFuseAndVectorize when types not match.

### PR DESCRIPTION
**Pull request by hanhanW**
The sandbox approach is designed for the cases that element types match.

There are timing issues in e2e_matmul_direct test suites compilation.
Because some vector.broadcast ops will be created in some cases. This is
not the area that we're targeting in sandbox codegen driver at this
moment, so we limit the use case of sandbox codegen driver based
approach.

### CODE REVIEW

File: iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 1]
```diff
@@ -344,6 +382,7 @@ static LogicalResult setARMRootConfig(FuncOp entryPointFn,
 static LogicalResult setRootConfig(
     FuncOp entryPointFn, linalg::ContractionOpInterface contractionOp,
     ArrayRef<LoopTilingAndDistributionInfo> tiledLoops) {
+  puts("contraction");
```
Comment (MaheshRavishankar) [inline iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp]: Debug code? Please remove.

File: iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp [hunk 2]
```diff
@@ -380,8 +419,21 @@ static LogicalResult setRootConfig(
     // number of loops when adding the pass to pass manager.
     // TODO(hanchung): Embed options into attributes, so we can control options
     // more heuristically.
-    return setX86SandboxRootConfig(entryPointFn, contractionOp,
-                                   workloadPerWorkgroup, vectorSize);
+    int lhsElemBitwidth = getElementTypeOrSelf(contractionOp.lhs().getType())
```
Comment (MaheshRavishankar) [inline iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp:421]: Do you need to check the bit width? How about just doing equality check on the element type itself.

**Review by MaheshRavishankar — APPROVED** [review]
Minor comments. Please address before submitting. Stamping for expediency.

File: iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp (hunk 2 above)
Comment (hanhanW) [reply iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp:421]: ah, good point.

**Pull request merged**